src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -670,6 +670,11 @@ public Integer deleteEntities(
             @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
             @RequestBody Set<UUID> entityKeyIds,
             @RequestParam( value = TYPE ) DeleteType deleteType ) {
+
+        if ( entityKeyIds.size() == dgm.getEntitySetSize( entitySetId ) ) {
+            return deleteAllEntitiesFromEntitySet( entitySetId, deleteType );
```

Comment:
This will probably work in pretty much every case, but it's theoretically possible that `entityKeyIds` could include an entityKeyId that was already deleted, meaning it wouldn't be included in the entity set size computed from DataGraphManager, which would result in extra entities being unintentionally deleted

Reply:
Oof, yeah you're totally right.
In that case though, I don't see how we could fix this aside from enforcing a max size on `entityKeyIds` which could still result in being able to remove an entire entity set through here but wouldn't be a big of deal.

Reply:
I believe the default is 128MB which would be like 8 million UUIDs

Reply:
I would go with enforcing too or doing it in chunks